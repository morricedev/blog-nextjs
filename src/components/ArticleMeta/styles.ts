import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkGray};
    font-style: italic;

    .categories span::after {
      content: ', ';
    }

    .categories span:last-child::after {
      content: '';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.3s ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
