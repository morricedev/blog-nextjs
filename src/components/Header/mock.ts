export default {
  blogName: 'Morrice',
  blogDescription:
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Nulla ad doloribus omnis eaque obcaecati magnam cum sequi ab ducimus nemo officia illo.',
  logo: 'assets/images/teste.jpg',
};
