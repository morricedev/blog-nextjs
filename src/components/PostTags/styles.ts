import styled, { css } from 'styled-components';

export const Wrapper = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;

    span {
      margin: 0 0 0 0.5rem;

      &::after {
        content: ', ';
      }

      &:last-child::after {
        content: '';
      }
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.3s ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
