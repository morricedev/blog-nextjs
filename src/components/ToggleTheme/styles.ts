import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  position: fixed;
  top: 2rem;
  right: 2rem;
  z-index: 2;
`;

export const Label = styled.label`
  position: relative;
  display: inline-block;
  width: 5rem;
  height: 2.5rem;
  line-height: 0;
  font-size: 0;
  overflow: hidden;
  color: transparent;
`;

export const Input = styled.input`
  ${({ theme }) => css`
    appearance: none;
    opacity: 0;
    width: 0;
    height: 0;

    &:checked + ${Slider} {
      background: ${theme.colors.secondary};

      &:before {
        transform: translateX(1.9rem);
        background: white;
      }
    }

    &:focus + ${Slider} {
      border: 1px solid blue;
    }
  `}
`;

export const Slider = styled.span`
  ${({ theme }) => css`
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: ${theme.colors.darkGray};
    transition: all 300ms ease-in-out;
    border-radius: 2rem;
    border: 1px solid transparent;

    &:before {
      content: '';
      position: absolute;
      height: 2.1rem;
      width: 2.1rem;
      left: 0.4rem;
      bottom: 0.1rem;
      background: ${theme.colors.primary};
      border-radius: 50%;
      transition: all 300ms ease-in-out;
    }
  `}
`;
