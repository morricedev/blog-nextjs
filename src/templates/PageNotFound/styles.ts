import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    height: 80vh;
    margin: 0 auto;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    text-align: center;

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.3s ease-in-out;

      &:hover {
        filter: brightness(80%);
      }
    }
  `}
`;
