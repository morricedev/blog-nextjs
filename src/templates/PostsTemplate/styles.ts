import styled, { css } from 'styled-components';

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    padding: 0 ${theme.spacings.large};
    margin: ${theme.spacings.large} 0;
    display: flex;
    justify-content: center;
  `}
`;

export const SearchContainer = styled.div`
  ${({ theme }) => css`
    margin: 0 auto;
    margin-bottom: ${theme.spacings.xlarge};
    padding: 0 ${theme.spacings.large};
    max-width: 120rem;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
  `}
`;

export const SearchInput = styled.input`
  ${({ theme }) => css`
    padding: 0.5rem ${theme.spacings.small};

    &:disabled {
      outline: 0.1rem solid ${theme.colors.secondary};
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    border: none;
    padding: ${theme.spacings.small} ${theme.spacings.large};
    cursor: pointer;

    &:disabled {
      background: ${theme.colors.darkGray};
    }
  `}
`;
