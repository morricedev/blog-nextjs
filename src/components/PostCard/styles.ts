import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      margin: ${theme.spacings.small} 0;
    }

    a {
      text-decoration: none;
      color: inherit;
      transition: color 0.3s ease-in-out;
    }

    &:hover {
      a {
        color: ${theme.colors.secondary};
      }

      img {
        opacity: 0.8;
      }
    }
  `}
`;

export const Cover = styled.img`
  max-width: 100%;
  height: 20rem;
  transition: opacity 0.3s ease-in-out;
`;

export const Excerpt = styled.p``;
