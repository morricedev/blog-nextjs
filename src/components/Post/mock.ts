import { PostProps } from '.';
import { data } from '../../api/data.json';
import mock from '../HtmlContent/mock';

const { title, excerpt, cover, author, categories, createdAt } = data.posts[0];

export default {
  title,
  excerpt,
  cover,
  content: mock,
  author,
  categories,
  createdAt,
} as PostProps;
