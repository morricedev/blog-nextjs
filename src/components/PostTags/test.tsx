import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PostTags, PostTagsProps } from '.';
import mock from './mock';

const props: PostTagsProps = mock;

describe('<PostTags />', () => {
  it('should not render tags', () => {
    renderTheme(<PostTags tags={undefined} />);

    expect(screen.queryByRole('link')).not.toBeInTheDocument();
  });

  it('should render two tags', () => {
    renderTheme(<PostTags {...props} />);

    expect(screen.getAllByRole('link')).toHaveLength(2);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<PostTags {...props} />);

    expect(container).toMatchSnapshot();
  });
});
