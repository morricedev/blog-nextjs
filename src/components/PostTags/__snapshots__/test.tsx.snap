// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTags /> should match snapshot 1`] = `
.c0 {
  margin: 2.4rem 0;
}

.c0 span {
  margin: 0 0 0 0.5rem;
}

.c0 span::after {
  content: ', ';
}

.c0 span:last-child::after {
  content: '';
}

.c0 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

<div>
  <p
    class="c0"
  >
    tags:
    <span>
      <a
        href="/tag/use-state"
      >
        useState
      </a>
    </span>
    <span>
      <a
        href="/tag/use-effect"
      >
        useEffect
      </a>
    </span>
  </p>
</div>
`;
